src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5089,7 +5083,7 @@ static QVariant fcnTestGeomOverlayCrosses( const QVariantList &values, const Qgs
 
 static QVariant fcnGeomOverlayEquals( const QVariantList &values, const QgsExpressionContext *context, QgsExpression *parent, const QgsExpressionNodeFunction * )
 {
-  return executeGeomOverlay( values, context, parent, false, indexedFilteredOverlay<&QgsGeometry::equals> );
+  return executeGeomOverlay( values, context, parent, false, indexedFilteredOverlay<&QgsGeometry::equals>, false, 0.01  ); //grow amount should adapt to current units
```

Comment:
re the comment, do the units matter here? I think we just want to grow it "slightly", where "slightly" could be `layer->geometryOptions()->geometryPrecision()` (or fallback to something like 10^-8 if `qgsDoubleNear( precision, 0.0 )`).